### ISSUE TITLE
 Desktop- IOU - Notification is only received for the first edit of IOU request in workspace

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#28520]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
**Issue found when executing PR** https://github.com/Expensify/App/pull/27299
## Action Performed:
1. Launch New Expensify app.
2. Create a workspace with admin and member.
3. [Workspace member] Send a money request to the workspace chat.
4. [Workspace admin] On desktop app, send a reply in thread to the money request.
5. [Workspace admin] Open other chat like Concierge and background the app.
6. [Workspace member] Open the IOU request and edit the amount, and then edit other fields.
Note that

## Expected Result:
The workspace admin on desktop app will receive notification for each edit on the IOU request (amount, description, date etc)

## Actual Result:
The workspace admin on desktop app only receives notification for the first IOU request edit

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** 1.3.75-3

**Reproducible in staging?:** Yes

**Reproducible in production?:** No

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/78819774/ccd54c89-0f10-447a-8f80-c93a8216930b

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by chiragsalian** [comment]
i wasn't able to tackle this today. I'll continue it on Monday if it hasn't been tackled yet.

**Comment by mountiny** [comment]
This is a new feature, so I don think its valid blocker, I have also assigned Andrew coming from the original PR

**Comment by arosiclair** [comment]
Can't reproduce this on staging **v1.3.75-8**.

@lanitochka17 it looks like step 4 is incorrect. The admin needs to open the expense and reply in the the expense thread for the notifications to display. NOT in the report thread.

Closing this out. Please re-open/comment if anything else comes up

**Comment by lanitochka17** [comment]
Issue reproducible on the latest build 1.4.9-2

https://github.com/Expensify/App/assets/78819774/f06b4d23-0307-4f56-ba6a-74199bfdf77e

**Comment by arosiclair** [comment]
@lanitochka17 this test isn't correct. It's the same issue as described [above](https://github.com/Expensify/App/issues/28520#issuecomment-1742593741). 

Employee needs to comment **directly on the expense** before they can receive notifications for the expense being edited. The comment you added here is on the _report_:

![image](https://github.com/Expensify/App/assets/6833644/a57d6e17-bf4c-4a64-a4d0-b9e24f17d89f)

What regression test are you following? It probably needs to be updated

**Comment by arosiclair** [comment]
@lanitochka17 bump ^

**Comment by sakluger** [comment]
@lanitochka17 assigning you since we're waiting on your response.

**Comment by mvtglobally** [comment]
This issue found when executing PR 
#27299. 
We will double check if any TR steps need updating

**Comment by arosiclair** [comment]
@mvtglobally @lanitochka17 this test [IOU Actions - Edit Amount](https://expensify.testrail.io/index.php?/cases/view/2730113) is incorrect. Step 2 should be: 

> Navigate to the money request details, open the expense details and send a message (this will subscribe the user to notifications)

I also see that these tests do not have that step and they should have them
1. [IOU Actions - Edit Merchant](https://expensify.testrail.io/index.php?/cases/view/2730114)
2. [IOU Actions - Edit Date](https://expensify.testrail.io/index.php?/cases/view/2761747)
3. [IOU Actions - Edit Description](https://expensify.testrail.io/index.php?/cases/view/2761748)

**Comment by sakluger** [comment]
@arosiclair just to confirm, is this even a bug, or are you saying that it's expected behavior?

**Comment by arosiclair** [comment]
> @arosiclair just to confirm, is this even a bug, or are you saying that it's expected behavior?

Yeah it's expected behavior. Without commenting on an expense, you don't get notifications for edits on the expense (same way it works for commenting on threads).

**Comment by arosiclair** [comment]
@mvtglobally it still doesn't look like those tests we're updated

**Comment by arosiclair** [comment]
@mvtglobally bump also [in slack](https://expensify.slack.com/archives/C9YU7BX5M/p1703793288358339)

**Comment by isagoico** [comment]
@arosiclair after a quick review we modified the test cases to the following: 
- Edited step 2 in  TC [IOU Actions - Edit Amount](https://expensify.testrail.io/index.php?/cases/view/2730113) to be clearer that the message should be sent in the expense/transaction details conversation for the user to be subscribed to notifications. 

_Before_: Navigate to the request details conversation and send a message (this will subscribe the user to notifications)
_**After**_: Navigate to the expense/transaction details conversation and send a message (this will subscribe the user to notifications)

For the consequent TCs, to avoid repeating the step of creating a new request and subscribing the user each time, I added a note that it should be executed in the same money request as in the first TC in the IOU Actions series > [IOU Actions - Edit Amount](https://expensify.testrail.io/index.php?/cases/view/2730113). This would save us a bit of time when executing these TCs. Let me know if it's valuable to create a new request and subscribing to each for each edit action TC. 

Additionally, we will remind our testers the difference between the 3 conversation types (1:1 or parent conversation, report conv and request conv).
